### PR TITLE
View Dependent Details

### PR DESCRIPTION
**Pull request by tiagograf-AOT**
# Fixes or Implements [AB#9243](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/9243)

* [x] Enhancement
* [ ] Bug
* [ ] Documentation

## Description

View Dependent Card with Covid results.

## Testing

* [ ] Unit Tests Updated
* [ ] Functional Tests Updated
* [x] Not Required

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

YES

### Browsers Tested

* [x] Chrome
* [ ] Safari
* [ ] Edge
* [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant.  Include any specialized deployment steps here.  

</br>
</br>
</br>
___
  
## Reviewer Guidelines

A reviewer should examine the PR if the Status above indicates the PR is **Ready**.
Once ready, the reviewer should read the information below as well as supporting information like the Azure ticket to ensure they have a good understanding of the request.

The reviewer should then:

1. Is the PR template complete and accurate?

    Confirm by reading all sections below.

1. Confirm the merge branch

    A normal feature/bug branch should merge into the Dev branch.  Exceptions for hotfix and sprint merges do occur and the target would then be Master.

1. Do the code commits have an appropriate comment and include linkage to the azure ticket by including a reference to "AB#nnnn"?

1. Review the Code Changes

    1. Do the changes match the description?

        The code commits should reflect the documented description and nothing more.  If additional changes are included the reviewer should call these out and request that the developer document the changes in Azure and in the description or remove the changed code.

    1. useful tips

        * Any obvious errors or omissions?
        * Does the code match the style of other similar work?
        * Is the code easy to understand?
        * Is the code well documented?
        * DB Migrations are appropriate?

If the Answer to any of the questions above is No, then a request for change should be the result of the review.

### CODE REVIEW

File: Apps/WebClient/src/ClientApp/src/components/dependentCard.vue [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+            })
+            .finally(() => {
+                this.isLoading = false;
+            });
+    }
+
+    private getReport(labOrder: LaboratoryOrder) {
+        let labResult = labOrder.labResults[0];
+        this.laboratoryService
+            .getReportDocument(labResult.id, this.user.hdid)
```
Comment (leobcgov) [inline Apps/WebClient/src/ClientApp/src/components/dependentCard.vue:96]: should we pass the dependent's hdid or the this.user.hdid here?

File: Apps/WebClient/src/ClientApp/src/components/dependentCard.vue [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+                this.addError(
+                    ErrorTranslator.toBannerError(
+                        "Download Laboratory Report Error",
+                        err
+                    )
+                );
+            });
+    }
+
+    private formatDate(date: StringISODate): string {
```
Comment (leobcgov) [inline Apps/WebClient/src/ClientApp/src/components/dependentCard.vue:116]: I think we have this refactored somewhere on vue side, we can just reuse the existing one.

File: Apps/WebClient/src/ClientApp/src/components/modal/verifySMS.vue [hunk 3]
```diff
@@ -186,7 +187,7 @@ export default class VerifySMSComponent extends Vue {
             resendTime = this.smsResendDateTime;
         }
         resendTime = resendTime.add(this.config.timeouts.resendSMS * 60 * 1000);
-        return resendTime.diff(new DateWrapper());
+        return resendTime.diff(new DateWrapper()).milliseconds;
```
Comment (leobcgov) [inline Apps/WebClient/src/ClientApp/src/components/modal/verifySMS.vue:190]: is it a bug? if yes, should we get a separate bug to fix it?

File: Apps/WebClient/src/ClientApp/src/models/dateWrapper.ts [hunk 4]
```diff
@@ -188,10 +188,10 @@ export class DateWrapper {
      * Difference between two Dates in milliseconds
      * @param other Other date to compare to
      * @param unit (optional) unit of comparision (year, hour, minute, second...)
-     * @returns the difference in milliseconds
+     * @returns the difference with the Duration object
      */
-    public diff(other: DateWrapper, unit?: DurationUnit): number {
-        return this.internalDate.diff(other.internalDate, unit).milliseconds;
+    public diff(other: DateWrapper, unit?: DurationUnit): Duration {
```
Comment (leobcgov) [inline Apps/WebClient/src/ClientApp/src/models/dateWrapper.ts:193]: This fix seems not to be a part of the scope of this PR.

**Review by leobcgov — CHANGES_REQUESTED** [review]
Can we see some screenshots if any?
Builds failed for both Admin and WebClient, it may be not relevant but preventing approval.

File: Apps/WebClient/src/ClientApp/src/components/modal/verifySMS.vue (hunk 3 above)
Comment (sslaws) [reply Apps/WebClient/src/ClientApp/src/components/modal/verifySMS.vue:190]: Leo:  Its ok to fix any bug while working on a ticket.  Just need to document it in the PR or possibly what you're indicating create a bug and reference it in the PR as well.

**Review by sslaws — CHANGES_REQUESTED** [review]
Questions as per Leo:  ID being passed in seems wrong.

File: Apps/WebClient/src/ClientApp/src/models/dateWrapper.ts (hunk 4 above)
Comment (tiagograf-AOT) [reply Apps/WebClient/src/ClientApp/src/models/dateWrapper.ts:193]: not a bug just made the method more useful.

File: Apps/WebClient/src/ClientApp/src/components/modal/verifySMS.vue (hunk 3 above)
Comment (tiagograf-AOT) [reply Apps/WebClient/src/ClientApp/src/components/modal/verifySMS.vue:190]: not a bug

File: Apps/WebClient/src/ClientApp/src/components/dependentCard.vue (hunk 1 above)
Comment (tiagograf-AOT) [reply Apps/WebClient/src/ClientApp/src/components/dependentCard.vue:96]: good point, must be the dependent id!

File: Apps/WebClient/src/ClientApp/src/components/dependentCard.vue (hunk 2 above)
Comment (tiagograf-AOT) [reply Apps/WebClient/src/ClientApp/src/components/dependentCard.vue:116]: i'll look into that

Comment (tiagograf-AOT) [reply Apps/WebClient/src/ClientApp/src/components/dependentCard.vue:116]: @leobcgov can you point me to where? i couldn't find it and i was unable to create a solution that didn't require a local format method in the vue file. even if I move the method to a separate file it always requires a local function to call the external method.

Comment (leobcgov) [reply Apps/WebClient/src/ClientApp/src/components/dependentCard.vue:116]: @tiagograf-AOT I may be wrong as we may have a different formatDate function. We can leave it as is then refactor it in the future then

**Review by leobcgov — APPROVED** [review]

**Review by sslaws — APPROVED** [review]

**Pull request merged**